### PR TITLE
feat: add layer2 crosschain module

### PR DESCRIPTION
**Pull request by zakir-code**

### CODE REVIEW

File: x/layer2/types/genesis.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package types
+
+import (
+	crosschaintypes "github.com/functionx/fx-core/v6/x/crosschain/types"
+)
+
+func DefaultGenesisState() *crosschaintypes.GenesisState {
+	params := crosschaintypes.DefaultParams()
+	params.GravityId = "fx-layer2-bridge"
+	params.AverageExternalBlockTime = 2_000
```
Comment (fx0x55) [inline x/layer2/types/genesis.go:10]: 2s is probably too short

File: x/layer2/module.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+// QuerierRoute implements app module
+func (am AppModule) QuerierRoute() string { return "" }
+
+// LegacyQuerierHandler returns no sdk.Querier
+func (am AppModule) LegacyQuerierHandler(*codec.LegacyAmino) sdk.Querier {
+	return nil
+}
+
+// RegisterServices registers module services.
+func (am AppModule) RegisterServices(cfg module.Configurator) {}
```
Comment (fx0x55) [inline x/layer2/module.go:111]:
```suggestion
func (am AppModule) RegisterServices(_ module.Configurator) {}
```

File: x/layer2/types/genesis.go (hunk 1 above)
Comment (zakir-code) [reply x/layer2/types/genesis.go:10]: We are not sure about layer2's block time.

Comment (fx0x55) [reply x/layer2/types/genesis.go:10]: How about setting it high to avoid a timeout, such as 30 minutes @zakir-code

**Review by fx0x55 — APPROVED** [review]
LGTM

**Pull request merged**